src/Abc.Zebus.Persistence.RocksDb/RocksDbStorage.cs

```diff
@@ -0,0 +1,242 @@
+    /// -------------------------------------------------------------------
+    /// </summary>
+    public class RocksDbStorage : IStorage, IDisposable
+    {
+        private static readonly int _guidLength = Guid.Empty.ToByteArray().Length;
+
+        private readonly ConcurrentDictionary<MessageId, bool> _outOfOrderAcks = new ConcurrentDictionary<MessageId, bool>();
+
+        private RocksDbSharp.RocksDb _db;
+        private readonly string _databaseDirectoryPath;
+        private ColumnFamilyHandle _messagesColumnFamily;
+        private ColumnFamilyHandle _peersColumnFamily;
+        private ColumnFamilyHandle _acksColumnFamily;
+
+        [DefaultConstructor]
+        public RocksDbStorage()
+            : this(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "database"))
+        {
+        }
+
+        public RocksDbStorage(string databaseDirectoryPath)
+        {
+            _databaseDirectoryPath = databaseDirectoryPath;
+        }
+
+        public int PersistenceQueueSize { get; } = 0;
+
+        public void Start()
+        {
+            var options = new DbOptions().SetCreateIfMissing()
+                                         .SetCreateMissingColumnFamilies()
+                                         .SetMaxBackgroundCompactions(4)
+                                         .SetMaxBackgroundFlushes(2)
+                                         .SetBytesPerSync(1024 * 1024);
+
+            var columnFamilies = new ColumnFamilies
+            {
+                { "Messages", ColumnFamilyOptions() },
+                { "Peers", ColumnFamilyOptions() },
+                { "Acks", ColumnFamilyOptions() }
+            };
+
+            _db = RocksDbSharp.RocksDb.Open(options, _databaseDirectoryPath, columnFamilies);
+
+            _messagesColumnFamily = _db.GetColumnFamily("Messages");
+            _peersColumnFamily = _db.GetColumnFamily("Peers");
+            _acksColumnFamily = _db.GetColumnFamily("Acks");
+
+            ColumnFamilyOptions ColumnFamilyOptions() => new ColumnFamilyOptions().SetCompression(CompressionTypeEnum.rocksdb_no_compression)
+                                                                                  .SetLevelCompactionDynamicLevelBytes(true)
+                                                                                  .SetArenaBlockSize(16 * 1024);
+
+            LoadAllOutOfOrderAcks();
+        }
+
+        public void Stop() => Dispose();
+
+        public void Dispose() => _db?.Dispose();
+
+        public Task Write(IList<MatcherEntry> entriesToPersist)
+        {
+            foreach (var entry in entriesToPersist)
+            {
+                var key = CreateKeyBuffer(entry.PeerId);
+                FillKey(key, entry.PeerId, entry.MessageId.GetDateTime().Ticks, entry.MessageId.Value);
+                if (entry.IsAck)
+                {
+                    // Ack
+                    var bytes = _db.Get(key, _messagesColumnFamily);
```

Comment:
This is a read for every ack. I feel like this is far from ideal for a high throughput usage considering RocksDB has a LSM storage. But I guess it should work for most usage cases and it is much simpler like that. I wish there was a `Exist` method in `RocksDb` though.

Other alternatives:
- Keep unacked message ids in-memory (not scalable).
- Use a write-only approach like in the Cassandra persistence (too complex).